### PR TITLE
Add undefined state for missing commit status

### LINKED ISSUE #59: NoMethodError: undefined method `[]' for nil:NilClass
View details in Rollbar: [https://rollbar.com/yaroslavtupitskyi/ProjectR/items/18/](https://rollbar.com/yaroslavtupitskyi/ProjectR/items/18/)

```

NoMethodError: undefined method `[]' for nil:NilClass
  File "/home/deploy/releases/20161125133846/app/jobs/application_job.rb", line 12, in status
  File "/home/deploy/releases/20161125133846/app/jobs/rebaser_job.rb", line 11, in perform
  File "/home/deploy/releases/20161125133846/app/controllers/rebases_controller.rb", line 34, in pull_request
  File "/home/deploy/releases/20161125133846/app/controllers/rebases_controller.rb", line 11, in create
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/basic_implicit_render.rb", line 4, in send_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/base.rb", line 188, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/rendering.rb", line 30, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/callbacks.rb", line 20, in block in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 126, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 506, in block (2 levels) in compile
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 455, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 101, in __run_callbacks__
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 750, in _run_process_action_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 90, in run_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/callbacks.rb", line 19, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/rescue.rb", line 20, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/instrumentation.rb", line 32, in block in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications.rb", line 164, in block in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications/instrumenter.rb", line 21, in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications.rb", line 164, in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/instrumentation.rb", line 30, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/params_wrapper.rb", line 248, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activerecord-5.0.0/lib/active_record/railties/controller_runtime.rb", line 18, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/base.rb", line 126, in process
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionview-5.0.0/lib/action_view/rendering.rb", line 30, in process
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal.rb", line 190, in dispatch
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal.rb", line 262, in dispatch
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/routing/route_set.rb", line 50, in dispatch
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/routing/route_set.rb", line 32, in serve
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/journey/router.rb", line 39, in block in serve
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/journey/router.rb", line 26, in each
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/journey/router.rb", line 26, in serve
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/routing/route_set.rb", line 725, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rack-2.0.1/lib/rack/etag.rb", line 25, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rack-2.0.1/lib/rack/conditional_get.rb", line 38, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rack-2.0.1/lib/rack/head.rb", line 12, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rack-2.0.1/lib/rack/session/abstract/id.rb", line 222, in context
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rack-2.0.1/lib/rack/session/abstract/id.rb", line 216, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/cookies.rb", line 613, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/callbacks.rb", line 38, in block in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 97, in __run_callbacks__
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 750, in _run_call_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 90, in run_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/callbacks.rb", line 36, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/remote_ip.rb", line 79, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rollbar-2.12.0/lib/rollbar/middleware/rails/rollbar.rb", line 24, in block in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rollbar-2.12.0/lib/rollbar.rb", line 735, in scoped
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rollbar-2.12.0/lib/rollbar/middleware/rails/rollbar.rb", line 22, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/debug_exceptions.rb", line 49, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/rollbar-2.12.0/lib/rollbar/middleware/rails/show_exceptions.rb", line 22, in call_with_rollbar
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_dispatch/middleware/show_exceptions.rb", line 31, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/railties-5.0.0/lib/rails/rack/logger.rb", line 36, in call_app
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/railties-5.0.0/lib/rails/rack/logger.rb", line 24, in block in call
  Fil

### LINKED ISSUE #63: NoMethodError: undefined method `[]' for nil:NilClass
View details in Rollbar: [https://rollbar.com/yaroslavtupitskyi/ProjectR/items/19/](https://rollbar.com/yaroslavtupitskyi/ProjectR/items/19/)

```

NoMethodError: undefined method `[]' for nil:NilClass
  File "/home/deploy/releases/20170105125254/app/jobs/application_job.rb", line 12, in status
  File "/home/deploy/releases/20170105125254/app/jobs/pusher_job.rb", line 13, in perform
  File "/home/deploy/releases/20170105125254/app/jobs/rebaser_job.rb", line 25, in rebase_pr
  File "/home/deploy/releases/20170105125254/app/jobs/rebaser_job.rb", line 10, in perform
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/execution.rb", line 34, in block in perform_now
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 126, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 506, in block (2 levels) in compile
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 455, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 448, in block (2 levels) in around
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 286, in block (2 levels) in halting
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/i18n-0.7.0/lib/i18n.rb", line 257, in with_locale
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/translation.rb", line 7, in block (2 levels) in <module:Translation>
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 391, in instance_exec
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 391, in block in make_lambda
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 285, in block in halting
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 447, in block in around
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 455, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 448, in block (2 levels) in around
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 286, in block (2 levels) in halting
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/logging.rb", line 24, in block (4 levels) in <module:Logging>
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications.rb", line 164, in block in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications/instrumenter.rb", line 21, in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/notifications.rb", line 164, in instrument
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/logging.rb", line 23, in block (3 levels) in <module:Logging>
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/logging.rb", line 44, in block in tag_logger
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/tagged_logging.rb", line 70, in block in tagged
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/tagged_logging.rb", line 26, in tagged
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/tagged_logging.rb", line 70, in tagged
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/logging.rb", line 44, in tag_logger
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/logging.rb", line 20, in block (2 levels) in <module:Logging>
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 391, in instance_exec
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 391, in block in make_lambda
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 285, in block in halting
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 447, in block in around
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 455, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 101, in __run_callbacks__
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 750, in _run_perform_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 90, in run_callbacks
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/execution.rb", line 33, in perform_now
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activejob-5.0.0/lib/active_job/execution.rb", line 16, in perform_now
  File "/home/deploy/releases/20170105125254/app/jobs/refresher_job.rb", line 9, in block in perform
  File "/home/deploy/releases/20170105125254/app/jobs/refresher_job.rb", line 7, in each
  File "/home/deploy/releases/20170105125254/app/jobs/refresher_job.rb", line 7, in perform
  File "/home/deploy/releases/20170105125254/app/controllers/rebases_controller.rb", line 30, in push
  File "/home/deploy/releases/20170105125254/app/controllers/rebases_controller.rb", line 11, in create
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/basic_implicit_render.rb", line 4, in send_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/base.rb", line 188, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/action_controller/metal/rendering.rb", line 30, in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/actionpack-5.0.0/lib/abstract_controller/callbacks.rb", line 20, in block in process_action
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 126, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 506, in block (2 levels) in compile
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 455, in call
  File "/home/deploy/shared/bundle/ruby/2.3.0/gems/activesupport-5.0.0/lib/active_support/callbacks.rb", line 101, in __run_callbacks__
  File "/home/deploy/shared/bund

### PR DESCRIPTION
**Pull request by Sky7hunder**
Close #63, close #59

### CODE REVIEW

File: spec/rails_helper.rb [hunk 1]
```diff
@@ -35,19 +35,11 @@
...
-
-  config.before(:each, js: true) do
     DatabaseCleaner.strategy = :truncation
   end
 
-  config.after(:suite) do
-    DatabaseCleaner.clean
+  config.around :each do |example|
+    DatabaseCleaner.cleaning { example.run }
   end
```
Comment (galulex) [inline spec/rails_helper.rb]: [here](https://github.com/DatabaseCleaner/database_cleaner#rspec-with-capybara-example) is the right config can you put it here and remove the garbage

File: app/jobs/application_job.rb [hunk 2]
```diff
@@ -9,7 +9,8 @@ def set_status(status, options = {})
 
   def status
     status = @rebase.user.github.status(@rebase.repo, @rebase.head)
-    status['statuses'][0]['state']
+    rebase_status = status['statuses'].find{|status| status['context'] == 'ProjectR' }.try(:[], 'state')
```
Comment (galulex) [inline app/jobs/application_job.rb]: do not use `try`. Use `&.dig` in this case

Comment (galulex) [inline app/jobs/application_job.rb]: Do not hardcode `strings` extract it to constant `STATUS`

File: app/jobs/application_job.rb [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
   include Rails.application.routes.url_helpers
 
+  APPLICATION_TITLE = 'ProjectR'.freeze
+  STATUSES = {
+    error: 'error',
+    pending: 'pending',
+    failure: 'failure',
+    success: 'success',
+    undefined: 'undefined'
+  }.freeze
```
Comment (galulex) [inline app/jobs/application_job.rb]: Really? Why not array?

Comment (galulex) [inline app/jobs/application_job.rb]: The statuses constant makes everything more complicated

Comment (Sky7hunder) [inline app/jobs/application_job.rb]: I'm not well understood you. Are you want me to make single constant for each status or array for all statuses? What the best solution for you?

Comment (galulex) [inline app/jobs/application_job.rb]: Remove STATUSES constant

File: circle.yml [hunk 4]
```diff
@@ -3,3 +3,6 @@ deployment:
     branch: master
     commands:
       - bundle exec cap production deploy
+machine:
+  ruby:
+    version: 2.3.1
```
Comment (galulex) [inline circle.yml]: why old ruby?

**Pull request merged**